dataService.js

```diff
@@ -0,0 +1,53 @@
+// Getting data from json files
+const categoryData = require('./src/data/categories.json');
+const productData = require('./src/data/products.json');
+
+// Maps
+let productMap = {};
+let categoryMap = {};
+let combinedProductMap = [];
+
+function getProducts() {
+  productData.products.forEach(product => {
+    productMap[product.id] = product;
+  });
+}
+
+function getCategories() {
+  categoryData.categories.forEach(category => {
+    categoryMap[category.id] = category.categoryName;
+  });
+}
+
+function combineProductsWithCategories() {
+
+  // Creating map of categories
+  getCategories();
+
+  productData.products.forEach(product => {
+    // Making deep copy to prevent changes in actual data
+    let productToBeReturned = JSON.parse(JSON.stringify(product));
```

Comment:
Do we actually need to parse and strigify?

Reply:
Replaced with `Object.assign({}, product)`